ui/src/pages/PreviewStarterPage/PreviewStarterPage.component.tsx

```diff
@@ -19,6 +26,25 @@ export function PreviewStarterPage() {
   const treeState = useTreeState('README.md');
   const { call, clearError, isLoading, errorMessage } = useApiCall();
 
+  // function references were introduced so that 'useCallback' can depend on them
+  const openFileRef = treeState.openFile;
+  const toggleDirRef = treeState.toggleDir;
+
+  const openMainScala = useCallback(() => {
```

Comment:
Is there a reason for using `useCallback`?
I would consider using `useEffect` here instead of creating a separate callback and calling it in another `useEffect`.

Reply:
That was some historical reason which is no longer valid. Good catch I have just removed it!